File: web-services/libraries/database/src/main/java/com/cryptocaddy/libraries/database/dao/UserExchangeRepository.java
package com.cryptocaddy.libraries.database.dao;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import javax.transaction.Transactional;
import java.util.List;

@Repository
@Transactional
public interface UserExchangeRepository extends CrudRepository<UserExchange, Long> {

    //find all exchanges for a user
    List<UserExchange> findByUser(User user);

    //find by the primary key
    UserExchange findById(Long Id);

    void removeById(Long Id);

}

File: web-services/libraries/database/src/main/java/com/cryptocaddy/libraries/database/dao/UserExchangeRepository.kt
package com.cryptocaddy.libraries.database.dao

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

import javax.transaction.Transactional

@Repository
@Transactional
interface UserExchangeRepository : CrudRepository<UserExchange, Long> {

    //find all exchanges for a user
    fun findByUser(user: User): List<UserExchange>

    //find by the primary key
    fun findById(Id: Long?): UserExchange

    fun removeById(Id: Long?)

}
